### ISSUE TITLE
DHL version

### ISSUE THREAD

**Issue by stijnb1234** [arjenbos/ha-postnl#11]
Love the concept of this integration, it works great for the PostNL packages! Could you look into a version for DHL too? DHL has https://my.dhlparcel.nl/ which also auto-detects all the packages sent to you.

**Comment by arjenbos** [comment]
Hi @stijnb1234 ,

Thanks!

In my dev environment I already have a working proof of concept for DHL, but I need to decide on my next steps. Some issues:
- Not sure if DHL should be added to this integration. I think it will go against the concept of Home Assistant. Every "brand" aka domain should have his own integration., for me it feels like I'm not following the concept of HA if I add DHL to the PostNL integration. So either, I rename this integration or create a separate integration for DHL.
- The lovelace doesn't support multiple entities. I already forked the lovelace, but I think I want to include the lovelace in the integration.
- Currently, all data is stored in the attributes of an entity, because this is required by the lovelace. This is not ideal (other official integrations do this as well so it's not that bad). I want to switch to the [Calendar entity](https://developers.home-assistant.io/docs/core/entity/calendar/) and add events. By doing this, people can create automatons based on the shipments. I have a work in progress pull request open for this: https://github.com/arjenbos/ha-postnl/pull/12.

So, TLDR: I first want to change some architectural stuff before I add DHL.

**Comment by stijnb1234** [comment]
I understand. Nice to hear you are working on it!

**Comment by stijnb1234** [comment]
> Hi @stijnb1234 ,
> 
> Thanks!
> 
> In my dev environment I already have a working proof of concept for DHL, but I need to decide on my next steps. Some issues:
> 
> * Not sure if DHL should be added to this integration. I think it will go against the concept of Home Assistant. Every "brand" aka domain should have his own integration., for me it feels like I'm not following the concept of HA if I add DHL to the PostNL integration. So either, I rename this integration or create a separate integration for DHL.
> * The lovelace doesn't support multiple entities. I already forked the lovelace, because I want to include the lovelace in the integration. By doing this, we do not relay on an archived repository.
> * Currently, all data is stored in the attributes of an entity, this is required by the lovelace. This is not ideal (other official integrations do this as well so it's not that bad). I want to switch to the [Calendar entity](https://developers.home-assistant.io/docs/core/entity/calendar/) and add events. By doing this, people can create automatons based on the shipments. I have a work in progress pull request open for this: [Add calendar sensor #12](https://github.com/arjenbos/ha-postnl/pull/12).
> 
> So, TLDR: I first want to change some architectural stuff before I add DHL.

My opinion on this: If you are planning to support more shipping providers, I recommend renaming it. If you only want to support PostNL and DHL, splitting it up is the way to go.

**Comment by Voxxie** [comment]
I would say split the integration, combine the lovelace cards.

**Comment by stijnb1234** [comment]
@arjenbos If you need help testing, please let me know.

**Comment by IIIdefconIII** [comment]
Woul be awesome!,
is it possible to do something likse this

```
  status: ne: delivered
  delivered: false
  first_letter: false
  header: false
```

**Comment by Voxxie** [comment]
Hi @arjenbos ,
Any update on this ? 
I am willing to help if needed. 

Can you maybe share the DHL-code that you already created?

**Comment by arjenbos** [comment]
> @arjenbos Hoi! Hoe staat dit er voor? We willen je graag helpen als je hem publiek maakt ;)

I didn't had/made time the recent months to work on this. But I still want to add DHL 😄 .

**Comment by Tezzlicious** [comment]
@arjenbos any news on this? Thanks!
